packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -108,29 +109,27 @@ private static boolean isFileOrAsset(Uri uri) {
     }
 
     private MediaSource buildMediaSource(
-        Uri uri, DataSource.Factory mediaDataSourceFactory, Context context) {
+        Uri uri, DataSource.Factory mediaDataSourceFactory, String formatHint, Context context) {
       int type = Util.inferContentType(uri.getLastPathSegment());
-      switch (type) {
-        case C.TYPE_SS:
-          return new SsMediaSource.Factory(
-                  new DefaultSsChunkSource.Factory(mediaDataSourceFactory),
-                  new DefaultDataSourceFactory(context, null, mediaDataSourceFactory))
-              .createMediaSource(uri);
-        case C.TYPE_DASH:
-          return new DashMediaSource.Factory(
-                  new DefaultDashChunkSource.Factory(mediaDataSourceFactory),
-                  new DefaultDataSourceFactory(context, null, mediaDataSourceFactory))
-              .createMediaSource(uri);
-        case C.TYPE_HLS:
-          return new HlsMediaSource.Factory(mediaDataSourceFactory).createMediaSource(uri);
-        case C.TYPE_OTHER:
-          return new ExtractorMediaSource.Factory(mediaDataSourceFactory)
-              .setExtractorsFactory(new DefaultExtractorsFactory())
-              .createMediaSource(uri);
-        default:
-          {
-            throw new IllegalStateException("Unsupported type: " + type);
-          }
+      if (type == C.TYPE_SS || "ss".equals(formatHint)) {
```

Comment:
what happens if the type is C.TYPE_DASH and the formatHint is "ss"? 
If the formatHint is a back up when type is not found, we should only use formatHint after finding the type is not supported or found.
If the formatHint is an override, then we should use formatHint when it is present in the arguments and ignore the type.

Reply:
Had to wait on my coworker to get back in to ask him what he wanted. We agreed overriding is probably best.